codec/src/main/java/io/netty/handler/codec/MessageToByteEncoder.java

```diff
@@ -46,7 +46,7 @@
 public abstract class MessageToByteEncoder<I> extends ChannelOutboundHandlerAdapter {
 
     private final TypeParameterMatcher matcher;
-    private final boolean preferDirect;
+    protected final boolean preferDirect;
```

Comment:
I wonder if instead of exposing this it would be preferable to just expose a method which can take it into account by default:

```
protected final ByteBuf allocateBuffer(ChannelHandlerContext ctx, ByteBuf msg) {
  return allocateBuffer(ctx, msg, preferDirect);
}
```